From 041a6c3ecab08035e46d82c649948cdc239a990c Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Fri, 26 Sep 2025 10:08:40 +0200
Subject: [PATCH 219/553] Mark QUnicodeTools as security-critical

The code implemenents the Unicode boundary finding algorithm, so it's
security-critical with the same rationale as QStringMatcher.

The header file contains only declarations, so is significant-only.

Task-number: QTBUG-135195
Pick-to: 6.8
Change-Id: I5d99fc25d50b639f2bef3a93e6fa83be13208bda
Reviewed-by: Ivan Solovev <ivan.solovev@qt.io>
(cherry picked from commit 067883abcf39b806fd0d1f79190c6d45cd6dd34f)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/text/qunicodetools.cpp | 1 +
 src/corelib/text/qunicodetools_p.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/corelib/text/qunicodetools.cpp b/src/corelib/text/qunicodetools.cpp
index 332f24773e5..14c611bdb5b 100644
--- a/src/corelib/text/qunicodetools.cpp
+++ b/src/corelib/text/qunicodetools.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:critical reason:data-parser
 
 #include "qunicodetools_p.h"
 
diff --git a/src/corelib/text/qunicodetools_p.h b/src/corelib/text/qunicodetools_p.h
index 06f31a5511e..463e317b7d3 100644
--- a/src/corelib/text/qunicodetools_p.h
+++ b/src/corelib/text/qunicodetools_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:header-decls-only
 
 #ifndef QUNICODETOOLS_P_H
 #define QUNICODETOOLS_P_H
-- 
2.51.2

